### PR TITLE
feat: top navigation

### LINKED ISSUE #12: Implement Top Navigation
Implement 3-level top navigation as on exiting site:

Dropdown on hover:
Screenshot 2023-03-15 at 11 21 18

Dropdown with intermediate categories:
Screenshot 2023-03-15 at 11 21 47

Current content:
Screenshot 2023-03-15 at 11 22 49

Out of scope: 
- language navigation
- search field

### PR DESCRIPTION
**Pull request by smiroslav**
Please always provide the [GitHub issue(s)](../issues) your PR is for, as well as test URLs where your change can be observed (before and after):

Fix #12 

Test URLs:
- Before: https://main--mammotome--hlxsites.hlx.page/en/
- After: https://12-topnavigation--mammotome--hlxsites.hlx.page/en/

What is not implemented and will be dealt with in another task:
- animations for navigation items in mobile and desktop view
- language switcher
- "Kontakt us" button

### CODE REVIEW

File: styles/styles.css [hunk 1]
```diff
@@ -313,6 +312,12 @@ main img {
   width: 24px;
 }
 
+@media (min-width: 1025px) {
```
Comment (solaris007) [inline styles/styles.css:315]: is this change nav related? it would affect all icons everywhere.

**Comment by solaris007** [comment]
Some differences in behavior / layout

Mobile

Screenshot 2023-03-27 at 16 24 26
- header height too low
- logo seems slightly too big
- hamburger vertical position off
- header doesn't stay when you scroll up
- logo does not change to icon when you scroll up
- header height does not shrink when you scroll up
- search icon missing

Screenshot 2023-03-27 at 16 26 54
- visible gap after opening navigation
- missing icons
- font seems smaller/different
- close icon missing

Desktop

Screenshot 2023-03-27 at 16 29 32
- header height too low
- logo position different
- main navigation items still have the down-arrow
- second levels should open on-mouse-over

Screenshot 2023-03-27 at 16 34 09
- second level title font differences
- third level font differences
- margin/paddings between items, but also titles, not as in original

File: blocks/header/header.css [hunk 2]
```diff
@@ -183,7 +114,142 @@ header nav .nav-sections ul > li > ul > li {
...
 }
 
-@media (min-width: 900px) {
+.icon-logo {
+  width: 200px;
+}
+
+
+
+@media (max-width: 1024px) {
```
Comment (solaris007) [inline blocks/header/header.css:123]: should try to eliminate max-width breakpoints and override where necessary in a min-width break-point

File: blocks/header/header.css [hunk 3]
```diff
@@ -183,7 +114,142 @@ header nav .nav-sections ul > li > ul > li {
...
+  header nav {
+    gap: 0 1.4em;
+  }
+
+  header nav .nav-hamburger-icon::after {
+    content: '\f0c9';
+    font-family: var(--mt-nav-button-icons-font-family);
+    color: var(--mt-header-color-nav-icons);
+    font-size: 20px;
```
Comment (solaris007) [inline blocks/header/header.css:132]: use an existing var or add one

File: blocks/header/header.css [hunk 4]
```diff
@@ -183,7 +114,142 @@ header nav .nav-sections ul > li > ul > li {
...
+  }
+
+  header nav[aria-expanded="true"] .nav-sections {
+    position: absolute;
+    right: 0;
+    width: 300px;
+    height: 100vh;
+    background-color: var(--mt-global-color-base-primary);
+    color: var(--mt-global-color-white);
+    font-size: var(--mt-font-size-100);
```
Comment (solaris007) [inline blocks/header/header.css:147]: don't use raw font size vars directly, but use intermediate var

File: blocks/header/header.css [hunk 5]
```diff
@@ -183,7 +114,142 @@ header nav .nav-sections ul > li > ul > li {
...
+  }
+
+  header nav[aria-expanded="true"] .nav-sections > ul > li[aria-expanded="true"] > ul {
+    position: absolute;
+    width: 300px;
+    right: 0;
+    top: 0;
+    height: 100%;
+    background-color: var(--mt-global-color-base-primary);
+    color: var(--mt-global-color-white);
```
Comment (solaris007) [inline blocks/header/header.css:161]: don't use raw vars directly, but use intermediate var

File: blocks/header/header.css [hunk 6]
```diff
@@ -183,7 +114,142 @@ header nav .nav-sections ul > li > ul > li {
...
+    font-size: var(--mt-font-size-100);
```
Comment (solaris007) [inline blocks/header/header.css:162]: don't use raw vars directly, but use intermediate var

File: blocks/header/header.css [hunk 7]
```diff
@@ -183,7 +114,142 @@ header nav .nav-sections ul > li > ul > li {
...
+    font-weight: var(--mt-font-weight-bolder);
+    letter-spacing: 3px;
+    padding: 10px;
+  }
+
+  header nav .nav-sections > ul > li.nav-multilevel[aria-expanded="true"] > ul > li > ul {
+    display: flex;
+    flex-direction: column;
+    background-color: var(--mt-header-menu-second-level-nav-color);
+    color: var(--mt-global-color-white);
```
Comment (solaris007) [inline blocks/header/header.css:193]: don't use raw vars directly, but use intermediate var

File: blocks/header/header.css [hunk 8]
```diff
@@ -183,7 +114,142 @@ header nav .nav-sections ul > li > ul > li {
...
+    text-transform: none;
+    text-align: initial;
+    row-gap: unset;
+    font-size: var(--mt-font-size-90);
```
Comment (solaris007) [inline blocks/header/header.css:197]: don't use raw vars directly, but use intermediate var

File: blocks/header/header.css [hunk 9]
```diff
@@ -183,7 +114,142 @@ header nav .nav-sections ul > li > ul > li {
...
+  header nav[aria-expanded="true"] .nav-sections > ul > li > ul > li{
+    padding-top: 10px;
+    padding-bottom: 10px;
+  }  
+  
+  header nav .nav-sections > ul > li.nav-multilevel[aria-expanded="true"] > ul > li {
+    text-transform: uppercase;
+    text-align: center;
+    color: var(--mt-header-menu-items-fint-color);
+    font-size: var(--mt-font-size-80);
```
Comment (solaris007) [inline blocks/header/header.css:183]: don't use raw vars directly, but use intermediate var

File: blocks/header/header.css [hunk 10]
```diff
@@ -222,9 +288,10 @@ header nav .nav-sections ul > li > ul > li {
 
   header nav .nav-sections ul {
     display: flex;
+    justify-content: flex-end;
     gap: 2em;
     margin: 0;
-    font-size: var(--mt-body-font-size-m);
+    font-size: var(--mt-font-size-90);
```
Comment (solaris007) [inline blocks/header/header.css:294]: don't use raw vars directly, but use intermediate var

File: blocks/header/header.css [hunk 11]
```diff
@@ -239,14 +306,62 @@ header nav .nav-sections ul > li > ul > li {
...
-    width: 200px;
     margin-top: 12px;
-    padding: 1em;
-    white-space: initial;
+    white-space: nowrap;
+    background-color: var(--mt-global-color-base-primary);
+    border-style: solid;
+    border-width: 5px;
+    border-color: var(--mt-header-menu-border-color);
+    color: var(--mt-global-color-white);
```
Comment (solaris007) [inline blocks/header/header.css:323]: don't use raw vars directly, but use intermediate var

File: blocks/header/header.css [hunk 12]
```diff
@@ -239,14 +306,62 @@ header nav .nav-sections ul > li > ul > li {
...
+
+  header nav .nav-sections > ul > li.nav-multilevel[aria-expanded="true"] > ul > li > ul > li{
+    padding: 10px;
+  }
+
+  header nav .nav-sections > ul > li.nav-multilevel[aria-expanded="true"] > ul > li > ul {
+    display: flex;
+    flex-direction: column;
+    background-color: var(--mt-header-menu-second-level-nav-color);
+    color: var(--mt-global-color-white);
```
Comment (solaris007) [inline blocks/header/header.css:353]: don't use raw vars directly, but use intermediate var

File: styles/styles.css [hunk 13]
```diff
@@ -15,6 +15,11 @@
   src: local("Arial");
 }
 
+@font-face {
```
Comment (solaris007) [inline styles/styles.css:18]: do we really need this font? which icon do you need? you could do inline SVG instead.

Comment (smiroslav) [reply styles/styles.css:18]: It is for symbols used in mobile view navigation. I used the same font used currently for that in mammotome.com/

Screenshot_2023-03-27_at_16_50_55

Comment (solaris007) [reply styles/styles.css:18]: i understand. so far we've been using inline SVG to save a request and inclusion of a font file, of which we may only use a fraction. better ideas welcome.

**Comment by smiroslav** [comment]
I have created a follow-up issue to address the remaining PR comments:
https://github.com/hlxsites/mammotome/issues/45

**Review by solaris007 — APPROVED** [review]
